.github/workflows/main.yml

```diff
@@ -0,0 +1,39 @@
+name: run script.py
+
+on:
+  schedule:
+    - cron: '0 */2 * * *' # Every Two Hours 
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: checkout repo content
+        uses: actions/checkout@v2 # checkout the repository content to github runner
+
+      - name: setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9' # install the python version needed
+          
+      - name: install python packages
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          
+      - name: execute py script # run main.py
+        run: python script.py
+          
+      - name: commit files
+        run: |
+          git add -A
+          git rev-list --objects --all | git cat-file --batch-check='%(objecttype) %(objectname) %(objectsize) %(rest)' | awk '/^blob/ {print substr($0,6)}' | sort --numeric-sort --key=2 -r | head -n 20
+          git diff-index --quiet HEAD || (git commit -a -m "update target" --allow-empty)
```

Comment:
You should commit files only if there are changes, not use empty commits. 

Also, the commit message us not good. Please give a better message.